pkg/gds/store/leveldb/leveldb.go

```diff
@@ -572,6 +575,166 @@ func (s *Store) DeleteCertReq(id string) (err error) {
+	defer iter.Release()
+	for iter.Next() {
+		peer := new(peers.Peer)
+		if err = proto.Unmarshal(iter.Value(), peer); err != nil {
+			return nil, err
+		}
+		pl = append(pl, peer)
+	}
+
+	if err = iter.Error(); err != nil {
+		return nil, err
+	}
+
+	return pl, nil
+}
+
+// CreatePeer using its PID (can't be nil) to create the LDB key and return the version.
+func (s *Store) CreatePeer(p *peers.Peer) (id string, err error) {
+	// The ID on a Peer is a uint64 representing the PID
+	// convert to string for a consistent interface across Create and Retrieve methods
+	key := s.peerKey(p.Key())
+
+	// Check to see if the Peer was previously created and deleted
+	var val []byte
+	if val, err = s.db.Get(key, nil); err != nil {
+		// New peer, not a tombstone
+		if err == leveldb.ErrNotFound {
+			// Initialize the first version
+			p.Metadata = &global.Object{
+				Key:       string(key),
+				Namespace: nsReplicas,
+			}
+			if err = s.updateVersion(p.Metadata); err != nil {
+				return "", fmt.Errorf("could not create object version: %s", err)
+			}
+
+			var data []byte
+			if data, err = proto.Marshal(p); err != nil {
+				return "", err
+			}
+
+			if err = s.db.Put(key, data, nil); err != nil {
+				return "", err
+			}
+		} else {
+			// TODO: Or just ignore other Get errors?
+			return "", fmt.Errorf("error when creating peer %s: %s", id, err)
+		}
+	}
+
+	// Peer exists, unmarshall it
+	np := &peers.Peer{}
+	if err = proto.Unmarshal(val, np); err != nil {
+		// TODO: if old peer can't be unmarshalled, just overwrite with the new data?
+		return "", fmt.Errorf("peer %d found in database but could not be unmarshalled", p.Id)
+	}
+
+	// Check if this is a tombstone version
+	if np.Deleted == "" {
+		// Peer exists but has not be deleted, can't create
+		return "", fmt.Errorf("cannot create %d, already exists", p.Id)
+	}
+	// Reincarnate peer from tombstone
+	// Update management timestamps, record metadata, and undelete
+	np.Modified = time.Now().Format(time.RFC3339)
+	if np.Created == "" {
+		np.Created = np.Modified
+	}
+	np.Deleted = ""
+
+	// TODO: do we have to worry about the Region or Name changing?
+
+	if err = s.updateVersion(np.Metadata); err != nil {
+		return "", fmt.Errorf("could not update object version: %s", err)
+	}
+
+	// Remarshall
+	var data []byte
+	if data, err = proto.Marshal(np); err != nil {
+		return "", err
+	}
+
+	// Put back to the database
+	if err = s.db.Put(key, data, nil); err != nil {
+		return "", err
+	}
+
+	return p.Key(), nil
+}
+
+// RetrievePeer returns a peer by it's stringified PID.
+func (s *Store) RetrievePeer(id string) (p *peers.Peer, err error) {
+	if id == "" {
+		return nil, ErrEntityNotFound
+	}
+
+	var val []byte
+	if val, err = s.db.Get(s.peerKey(id), nil); err != nil {
+		if err == leveldb.ErrNotFound {
+			return nil, ErrEntityNotFound
+		}
+		return nil, err
+	}
+
+	p = new(peers.Peer)
+	if err = proto.Unmarshal(val, p); err != nil {
+		return nil, err
+	}
+
```

Comment:
We should check for Tombstones here and return `ErrEntityNotFound` if it has been deleted. 

```go
if p.Deleted != "" {
	return nil, ErrEntityNotFound
}
```